storage/gcp/gcp.go

```diff
@@ -85,25 +96,95 @@ func New(ctx context.Context, cfg Config) (*Storage, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to create GCS storage: %v", err)
 	}
-
-	dbPool, err := spanner.NewClient(ctx, cfg.Spanner)
+	seq, err := newSpannerSequencer(ctx, cfg.Spanner)
 	if err != nil {
-		klog.Exitf("Failed to connect to Spanner: %v", err)
+		return nil, fmt.Errorf("failed to create Spanner sequencer: %v", err)
 	}
 
 	r := &Storage{
 		gcsClient: c,
 		projectID: cfg.ProjectID,
 		bucket:    cfg.Bucket,
-		dbPool:    dbPool,
 		objStore:  gcsStorage,
+		sequencer: seq,
 	}
+	// TODO(al): make queue options configurable:
+	r.queue = storage.NewQueue(time.Second, 256, r.sequencer.assignEntries)
 
-	if err := r.initDB(ctx); err != nil {
-		return nil, fmt.Errorf("failed to init DB: %v", err)
+	return r, nil
+}
+
+// tileSuffix returns either the empty string or a tiles API suffix based on the passed-in tile size.
+func tileSuffix(s uint64) string {
+	if p := s % 256; p != 0 {
+		return fmt.Sprintf(".p/%d", p)
 	}
+	return ""
+}
 
-	return r, nil
+// setTile stores a tile in GCS.
+//
+// The location to which the tile is written is defined by the tile layout spec.
+func (s *Storage) setTile(ctx context.Context, level, index uint64, tile [][]byte) error {
+	tileSize := uint64(len(tile))
+	klog.V(2).Infof("StoreTile: level %d index %x ts: %x", level, index, tileSize)
+	if tileSize == 0 || tileSize > 256 {
+		return fmt.Errorf("tileSize %d must be > 0 and <= 256", tileSize)
+	}
+	t := slices.Concat(tile...)
+
+	// Pass an empty rootDir since we don't need this concept in GCS.
```

Comment:
I'd just remove this, we can't even see where the emptyness is, this is only relevant if you've worked on a different implementation before.

Reply:
oops, that shouldn't be there, ta.